go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -121,7 +121,7 @@ func (qre *QueryExecutor) Execute() (reply *sqltypes.Result, err error) {
 		defer conn.Recycle()
 		switch qre.plan.PlanID {
 		case planbuilder.PlanPassDML:
-			if qre.tsv.qe.binlogFormat != connpool.BinlogFormatRow {
+			if !qre.tsv.qe.passthroughDMLs.Get() && (qre.tsv.qe.binlogFormat != connpool.BinlogFormatRow) {
```

Comment:
As discussed, vttablet should disallow any pass-through query for SBR, and we should add a separate override for vtqueryserver.

It may actually be better to fail early on this: QueryEngine calls VerifyMode on Open. If the flag is on, and we are not RBR, we can return an error there.

Reply:
I added the separate option.

Failing early seems like a nice to have but not necessary optimization.